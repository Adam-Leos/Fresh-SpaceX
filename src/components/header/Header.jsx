// @flow

import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './header.scss';

type PropType = {};
type StateType = {
  isMobileMenuActive: boolean,
}

class Header extends Component<PropType, StateType> {
  constructor() {
    super();

    this.state = {
      isMobileMenuActive: false,
    };
  }

  onHamburgerClick = () => this.toggleMobileMenuActivity();
  onHamburgerKeyUp = () => this.toggleMobileMenuActivity();

  toggleMobileMenuActivity = () => {
    this.setState({
      isMobileMenuActive: !this.state.isMobileMenuActive,
    });
  }

  render() {
    const activeClass = this.state.isMobileMenuActive ? 'active' : '';

    return (
      <header className="header">
        <div
          className={`${activeClass} header__hamburger`}
          onClick={this.onHamburgerClick}
          onKeyUp={this.onHamburgerKeyUp}
          role="menuitem"
          tabIndex="0"
        >
          <span className="header__hamburger-bar" />
          <span className="header__hamburger-bar" />
          <span className="header__hamburger-bar" />
        </div>
        <nav className={`${activeClass} header__nav`}>
          <Link className="header__link" to="/">Main</Link>
          <Link className="header__link" to="/rockets">Rockets</Link>
          <Link className="header__link" to="/launches">Launches</Link>
          <Link className="header__link" to="/contacts">Contacts</Link>
        </nav>
      </header>
    );
  }
}

export default Header;
