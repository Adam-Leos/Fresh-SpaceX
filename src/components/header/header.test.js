import React from 'react';
import Header from './Header.jsx';

describe('testing header component', () => {
  test('Header is matching snapshot', () => {
    const component = shallow(<Header />);

    expect(component).toMatchSnapshot();
  });

  test('Hamburger events working', () => {
    const component = shallow(<Header />);

    expect(component.find('.header__hamburger')).toHaveLength(1);
    expect(component.find('.header__hamburger').hasClass('active')).toEqual(false);

    component.find('.header__hamburger').simulate('click');
    expect(component.find('.header__hamburger').hasClass('active')).toEqual(true);
    expect(component.find('.header__hamburger')).toMatchSnapshot();

    component.find('.header__hamburger').simulate('keyUp');
    expect(component.find('.header__hamburger').hasClass('active')).toEqual(false);
    expect(component).toMatchSnapshot();
  });
});
