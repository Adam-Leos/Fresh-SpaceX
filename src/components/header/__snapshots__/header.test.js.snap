// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing header component Hamburger events working 1`] = `
<div
  className="active header__hamburger"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="menuitem"
  tabIndex="0"
>
  <span
    className="header__hamburger-bar"
  />
  <span
    className="header__hamburger-bar"
  />
  <span
    className="header__hamburger-bar"
  />
</div>
`;

exports[`testing header component Hamburger events working 2`] = `
<header
  className="header"
>
  <div
    className=" header__hamburger"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="menuitem"
    tabIndex="0"
  >
    <span
      className="header__hamburger-bar"
    />
    <span
      className="header__hamburger-bar"
    />
    <span
      className="header__hamburger-bar"
    />
  </div>
  <nav
    className=" header__nav"
  >
    <Link
      className="header__link"
      replace={false}
      to="/"
    >
      Main
    </Link>
    <Link
      className="header__link"
      replace={false}
      to="/rockets"
    >
      Rockets
    </Link>
    <Link
      className="header__link"
      replace={false}
      to="/launches"
    >
      Launches
    </Link>
    <Link
      className="header__link"
      replace={false}
      to="/contacts"
    >
      Contacts
    </Link>
  </nav>
</header>
`;

exports[`testing header component Header is matching snapshot 1`] = `
<header
  className="header"
>
  <div
    className=" header__hamburger"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="menuitem"
    tabIndex="0"
  >
    <span
      className="header__hamburger-bar"
    />
    <span
      className="header__hamburger-bar"
    />
    <span
      className="header__hamburger-bar"
    />
  </div>
  <nav
    className=" header__nav"
  >
    <Link
      className="header__link"
      replace={false}
      to="/"
    >
      Main
    </Link>
    <Link
      className="header__link"
      replace={false}
      to="/rockets"
    >
      Rockets
    </Link>
    <Link
      className="header__link"
      replace={false}
      to="/launches"
    >
      Launches
    </Link>
    <Link
      className="header__link"
      replace={false}
      to="/contacts"
    >
      Contacts
    </Link>
  </nav>
</header>
`;
